src/dfcx_scrapi/core/agent_assist.py

```diff
@@ -0,0 +1,286 @@
+"""Agent Assist Resource functions."""
```

Comment:
@hkhaitan1 this is really the first time we'll be accepting code from another CCAI product into the SCRAPI library, so I'd like to ensure we're keeping the code in a separate product root folder. The DFCX SCRAPI library, by its namesake, was always meant to extend the Dialogflow CX library. Although CCAI features like Agent Assist, Insights, Custom Voice, etc. are adjacent and utilized by CX, I'd like us to be cognizant of what we add here to not create confusion.

Please do the following:
- Create `agent_assist` folder under `src` (i.e. `src/agent_assist`)
- move `agent_assist.py` into the above folder (i.e. `src/agent_assist/agent_assist.py`)

One of the interesting things about the base API for Agent Assist is that it piggybacks off the Dialogflow ES (V2) source, which can be confusing for users.

Some Questions:
1. Have you been able to determine if we would ever need to logically create more than 1 Agent Assist Class to do all the necessary Agent Assist features?  
2. I see that you have many of the `create` functions but are missing others like `List`, `Update`, `Delete`, etc. Can we go ahead and add all of these Agent Assist CRUD functions to this class?

I want to be very careful that we're not mixing in Dialogflow ES features/functions into this particular class so we don't confuse users.

Reply:
I will move the code as requested.

Responses to your questions-

- Have you been able to determine if we would ever need to logically create more than 1 Agent Assist Class to do all the necessary Agent Assist features?
A: Based on the existing features in Agent Assist, majority configuration are at conversation profile level which is handled in this class. Also we do not need to create classes for individual AA features, instead going forward I will add methods to use individual features- like getting article suggestions, batch inference for model tuning, etc.

- I see that you have many of the create functions but are missing others like List, Update, Delete, etc. Can we go ahead and add all of these Agent Assist CRUD functions to this class?
A: Added CRUD operations.